projects/storefrontapp/server.ts

```diff
@@ -63,6 +66,9 @@ export function app(): express.Express {
     });
   });
 
+  server.use(handleCmsPageNotFoundErrorResponse(indexHtmlContent));
+  server.use(handleUnknownServerErrorResponse(indexHtmlContent));
+
```

Comment:
Strategic question: how do we plan to distribute our error handlers to customers?


- will we be adding them to their server.ts with schematics?
  - if yes, do we plan to put them one by one or somehow wrap them in one uber-error-handler, a.k.a. server.use(defaultServerErrorResponseHandlers()) // import defaultServerErrorResponseHandlers from @spartacus/setup/ssr?
    - thanks to this, over time we will be able to introduce changes/additions if needed - in existing apps. For example, we can easily add a temporaryRedirectErrorHandler (sending 302) if we really want to

Reply:
Yes, the idea is to provide it to fresh SPA apps via schematics. `uber-error-handler` is a tempting idea giving us possibilities for easier adjustments. The reason why I decided to introduce separate handlers was the question what if the customer wants to have our handler for unknown server errors, but they custom handler with an error page for 404? 
Actually, now I have an answer... the custom handler can be provided before ours. What do you think about that?

Reply:
[IMHO](https://blog.platis.dev/open-source) the most frequent use cases should be handled with a simple config, or super simple API. Ideally enabled implicitly by default. Medium-frequent use cases should require a tiny bit of coding, but proportionally to how often it is a use case. And for edge cases it's acceptable that customers might need to give up many OOTB default APIs and possibly write a lot of copy-pasted code + some customizations. (https://blog.platis.dev/open-source)

And in this case, in my opinion, default error handlers are good enough for most people. So, it should be a super simple and ergonomic API to use. Ideally implicit, but unfortunately, it's not possible because you have to put it explictly into the server.ts file.

If someone wants to add their error handlers - well, you nicely noticed that they can add their own above ours!! Nice catch!

If someone would like to remove one of our handlers at all, then they might want to remove our pack of error handlers. But for ergonomics, let's expose in our public APi all individuall handlers - so customers can pick and mix whatever they want, explicitly.